samples/boards/96b_argonkey/src/main.c

```diff
@@ -0,0 +1,335 @@
+{
+	return (val->val1 + (float)val->val2 / 1000000);
+}
+
+static int lsm6dsl_trig_cnt;
+#ifdef CONFIG_LSM6DSL_TRIGGER
+static void lsm6dsl_trigger_handler(struct device *dev,
+				    struct sensor_trigger *trig)
+{
+#ifdef ARGONKEY_TEST_LOG
+	char out_str[64];
+#endif
+	struct sensor_value accel_x, accel_y, accel_z;
+	struct sensor_value gyro_x, gyro_y, gyro_z;
+
+#if defined(CONFIG_LSM6DSL_EXT0_LIS2MDL)
+	struct sensor_value magn_x, magn_y, magn_z;
+#endif
+#if defined(CONFIG_LSM6DSL_EXT0_LPS22HB)
+	struct sensor_value press, temp;
+#endif
+
+	lsm6dsl_trig_cnt++;
+
+	sensor_sample_fetch_chan(dev, SENSOR_CHAN_ACCEL_XYZ);
+	sensor_channel_get(dev, SENSOR_CHAN_ACCEL_X, &accel_x);
+	sensor_channel_get(dev, SENSOR_CHAN_ACCEL_Y, &accel_y);
+	sensor_channel_get(dev, SENSOR_CHAN_ACCEL_Z, &accel_z);
+#ifdef ARGONKEY_TEST_LOG
+	sprintf(out_str, "accel (%f %f %f) m/s2", out_ev(&accel_x),
+						out_ev(&accel_y),
+						out_ev(&accel_z));
+	printk("TRIG %s\n", out_str);
+#endif
+
+	/* lsm6dsl gyro */
+	sensor_sample_fetch_chan(dev, SENSOR_CHAN_GYRO_XYZ);
+	sensor_channel_get(dev, SENSOR_CHAN_GYRO_X, &gyro_x);
+	sensor_channel_get(dev, SENSOR_CHAN_GYRO_Y, &gyro_y);
+	sensor_channel_get(dev, SENSOR_CHAN_GYRO_Z, &gyro_z);
+#ifdef ARGONKEY_TEST_LOG
+	sprintf(out_str, "gyro (%f %f %f) dps", out_ev(&gyro_x),
+						out_ev(&gyro_y),
+						out_ev(&gyro_z));
+	printk("TRIG %s\n", out_str);
+#endif
+
+#if defined(CONFIG_LSM6DSL_EXT0_LIS2MDL)
+	/* lsm6dsl magn */
+	sensor_sample_fetch_chan(dev, SENSOR_CHAN_MAGN_XYZ);
+	sensor_channel_get(dev, SENSOR_CHAN_MAGN_X, &magn_x);
+	sensor_channel_get(dev, SENSOR_CHAN_MAGN_Y, &magn_y);
+	sensor_channel_get(dev, SENSOR_CHAN_MAGN_Z, &magn_z);
+#ifdef ARGONKEY_TEST_LOG
+	sprintf(out_str, "magn (%f %f %f) gauss", out_ev(&magn_x),
+						 out_ev(&magn_y),
+						 out_ev(&magn_z));
+	printk("TRIG %s\n", out_str);
+#endif
+
+#endif
+#if defined(CONFIG_LSM6DSL_EXT0_LPS22HB)
+	/* lsm6dsl press/temp */
+	sensor_sample_fetch_chan(dev, SENSOR_CHAN_PRESS);
+	sensor_channel_get(dev, SENSOR_CHAN_PRESS, &press);
+
+	sensor_sample_fetch_chan(dev, SENSOR_CHAN_AMBIENT_TEMP);
+	sensor_channel_get(dev, SENSOR_CHAN_AMBIENT_TEMP, &temp);
+
+#ifdef ARGONKEY_TEST_LOG
+	sprintf(out_str, "press (%f) kPa - temp (%f) deg", out_ev(&press),
+							   out_ev(&temp));
+	printk("%s\n", out_str);
+#endif
+
+#endif
+}
+#endif
+
+void main(void)
+{
+	int cnt = 0;
+	char out_str[64];
+	static struct device *led0, *led1;
+	int i, on = 1;
+
+	led0 = device_get_binding(LED0_GPIO_CONTROLLER);
+	gpio_pin_configure(led0, LED0_GPIO_PIN, GPIO_DIR_OUT);
+	gpio_pin_write(led0, LED0_GPIO_PIN, 1);
+
+	led1 = device_get_binding(LED1_GPIO_CONTROLLER);
+	gpio_pin_configure(led1, LED1_GPIO_PIN, GPIO_DIR_OUT);
+
+	for (i = 0; i < 5; i++) {
+		gpio_pin_write(led1, LED1_GPIO_PIN, on);
+		k_sleep(200);
+		on = (on == 1) ? 0 : 1;
+	}
+
+	printk("ArgonKey test!!\n");
+
+#ifdef CONFIG_LPS22HB
+	struct device *baro_dev = device_get_binding("LPS22HB");
```

Comment:
Use the device name from dts rather than hardcoding the string. Do this for the other sensors too.